src/platform/Ligare/platform/feature_flag/feature_flag_router.py

```diff
@@ -1,7 +1,25 @@
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Generic, Sequence, TypeVar
 
 
-class FeatureFlagRouter(ABC):
+@dataclass(frozen=True)
+class FeatureFlag:
+    name: str
+    enabled: bool
+
+
+@dataclass(frozen=True)
+class FeatureFlagChange:
+    name: str
+    old_value: bool | None
```

Comment:
Is it necessary to have both old and new values here? What do we gain from storing the old value, and is there a risk if this gets out of sync? (Say, with multiple requests)

Reply:
We only use this in the HTTP PATCH response as a means to report back the old and new values. Because the status code is the same regardless of whether a value did change, this is a means for the caller to determine whether the value actually changed.